client/src/index.jsx

```diff
@@ -0,0 +1,206 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Price from './components/Price.jsx'
+import Photos from './components/Photos.jsx'
+import Description from './components/Description.jsx'
+// import Options from './components/Options.jsx'
+import Company from './components/Company.jsx'
+import Details from './components/Details.jsx'
+import Modal from './components/Modal.jsx'
+import AddToCart from './components/AddToCart.jsx'
+import AddToList from './components/AddToList.jsx'
+import ShareButtons from './components/ShareButtons.jsx'
+import MainPhoto from './components/MainPhoto.jsx'
+import QuantityMenu from './components/QuantityMenu.jsx'
+import StarRatingComponent from 'react-star-rating-component';
+import axios from 'axios';
+import primeLogo from '../dist/primelogo.png'
+ 
+class App extends React.Component {
+    constructor() {
+      super();
+      this.state = {
+        answers: 0,
+        description: '',
+        isHovered: false,
+        modalOpened: false,
+        showingModal: false,
+        photos: [],
+        prime: false,
+        mainphoto: '',
+        rating: 4.5,
+        selectedoption: '',
+        selectValue: '',
+        title: ''
+      }
+      this.getPhotos = this.getPhotos.bind(this);
+      this.hoverExit = this.hoverExit.bind(this);
+      this.hoverHandler = this.hoverHandler.bind(this);
+      this.handleModal = this.handleModal.bind(this);
+      this.hoverOptionHandler = this.hoverOptionHandler.bind(this);
+      this.handleQuantityChange = this.handleQuantityChange.bind(this);
+      this.showModal = this.showModal.bind(this);
+      this.hideModal = this.hideModal.bind(this);
+    }
+  
+    renderView() {
+    }
+
+    getMainProducts() {
+      axios.get('/api/mainProduct').then((data) => {
+        let mainProductData = data.data;
+        // console.log(mainProductData, 'line 48')
+        let test = Math.floor(Math.random() * 10);
+        mainProductData = mainProductData.filter(element => element.product_id === test)
```

Comment:
the method for filtering the product will need to happen on the backend for the different routes if we want all of our pages to display the same info based on an input id